File: src/main/java/com/trade/securities/application/service/CommodityTradingInfoService.java
package com.trade.securities.application.service;

import com.trade.securities.application.port.in.GetCommodityTradingInfoUseCase;
import com.trade.securities.application.port.out.LoadCommodityTradingInfoPort;
import com.trade.securities.domain.CommodityTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class CommodityTradingInfoService implements GetCommodityTradingInfoUseCase {

    private final LoadCommodityTradingInfoPort client;

    @Override
    public List<CommodityTradingInfo> getCommodityTradingInfo(String marketType, String basDd) {
        return client.loadCommodityTradingInfo(marketType, basDd);
    }
}

File: src/main/kotlin/com/trade/securities/application/service/CommodityTradingInfoService.kt
package com.trade.securities.application.service

import com.trade.securities.application.port.`in`.GetCommodityTradingInfoUseCase
import com.trade.securities.application.port.out.LoadCommodityTradingInfoPort
import com.trade.securities.domain.CommodityTradingInfo
import org.springframework.stereotype.Service

@Service
class CommodityTradingInfoService(
    private val loadCommodityTradingInfoPort: LoadCommodityTradingInfoPort
) : GetCommodityTradingInfoUseCase {

    override fun getCommodityTradingInfo(marketType: String, basDd: String): List<CommodityTradingInfo> {
        return loadCommodityTradingInfoPort.loadCommodityTradingInfo(marketType, basDd)
    }
}
